libcontainer/configs/validate/rootless.go

```diff
@@ -104,11 +94,24 @@ func rootlessMount(config *configs.Config) error {
 		// Check that the options list doesn't contain any uid= or gid= entries
 		// that don't resolve to root.
 		for _, opt := range strings.Split(mount.Data, ",") {
-			if strings.HasPrefix(opt, "uid=") && opt != "uid=0" {
-				return fmt.Errorf("cannot specify uid= mount options in rootless containers where argument isn't 0")
+			var uid int
+			n, err := fmt.Sscanf(opt, "uid=%d", &uid)
```

Comment:
We still need the `string.Contains` blocks from before, because now `gid=...` won't correctly be detected (the first `Sscanf` fails).